charts/alfresco-search-service/templates/deployment.yaml

```diff
@@ -89,26 +93,32 @@ spec:
             mountPath: {{ .Values.persistence.search.data.mountPath }}
             subPath: {{ .Values.persistence.search.data.subPath }}
           readinessProbe:
-            httpGet:
-              path: /solr/alfresco/admin/ping
-              port: {{ template "alfresco-search.containerPort" . }}
-              {{- if eq .Values.global.tracking.auth "secret" }}
-              httpHeaders:
-              - name: X-Alfresco-Search-Secret
-                value: "{{ .Values.global.tracking.sharedsecret }}"
-              {{- end }}
+            exec:
+              command:
+                - sh
+                - -c
+                - >-
+                  [
+                  $(
+                  curl -sH "X-Alfresco-Search-Secret: $SOLR_ALFRESCO_SECURECOMMS_SECRET" http://localhost:{{ template "alfresco-search.containerPort" . }}/solr/alfresco/admin/ping
+                  | xmllint --xpath '/response/str[@name="status"]/text()' -
```

Comment:
so the http response doesn't reflect the service status?

Reply:
not in case of 403. The exit code would still be 0 even though the server denied access